src/Mechanic.fs

```diff
@@ -45,7 +45,17 @@ let activate (context : Vscode.ExtensionContext) =
         Vscode.window.showWarningMessage("Mechanic couldn't be activated, Ionide-fsharp is required", [] |> ResizeArray)
         |> ignore
     | Some ext ->
-        ext.exports.ProjectLoadedEvent $ (state.Add) |> ignore
+        ext.exports.ProjectLoadedEvent $ (fun (project : Project) ->
+            let exist =
+                state
+                |> Seq.toList
+                |> List.exists(fun currentProject ->
+                    currentProject.Project = project.Project
+                )
+            // Only add the project if not already known
+            if not exist then
```

Comment:
I'm not sure about it - if we will use state to do something like detecting project for current file (like @forki suggested) then we need to have latest version of the project data. Maybe we should use Dictionary to make sure every project is only once in state?

Reply:
sorry merged it because I'm trying release process

Reply:
@Krzysztof-Cieslak Ah yes, I was working on the #9 and just think about this problem.

We indeed should change the structure to have only the latest version of the project. I can send this change with the #9 PR ?